lib/sdf-server/src/server/service/diagram/delete_component.rs

```diff
@@ -47,16 +54,71 @@ pub async fn delete_components(
         .await?;
         components.insert(component_id, component_still_exists);
 
+        let exists_on_head = components_existing_on_head
+            .get(&component_id)
+            .unwrap_or(&false);
```

Comment:
This feels really awkward. It's really unusual to see a reference to a bool. It looks like here is the only place that's using the output of `Component::exists_on_head`. Rather than returning a `HashMap<ComponentId, bool>` it would probably be more ergonomic to have it return a `HashSet<ComponentId>` of only the `ComponentId`s that do exist on `HEAD`. If it did that, then this check becomes `let exists_on_head = components_existing_on_head.contains(&component_id);` with no need for an extra unwrapping or defaults.